src/lib.rs

```diff
@@ -408,61 +408,155 @@
+//! you can use them via [raw methods](#raw-methods), for example:
 //!
 //! ```
 //! # use structopt::StructOpt;
 //!
 //! #[derive(StructOpt)]
-//! /// The help message that will be displayed when passing `--help`.
+//! #[structopt(about = "I am a program and I work, just pass `-h`")]
 //! struct Foo {
-//!   #[structopt(short)]
-//!   /// The description for the arg that will be displayed when passing `--help`.
+//!   #[structopt(short, help = "Pass `-h` and you'll see me!")]
 //!   bar: String
 //! }
-//! # fn main() {}
 //! ```
 //!
-//! If it is necessary or wanted to provide a more complex help message then the
-//! previous used ones, it could still be a good idea to distinguish between the
-//! actual help message a short summary. In this case `about` and `help` should
-//! only contain the short and concise form while the two additional arguments
-//! `long_about` and `long_help` can be used to store a descriptive and more in
-//! depth message.
-//!
-//! If both - the short and the long version of the argument - are present,
-//! the user can later chose between the short summary (`-h`) and the long
-//! descriptive version (`--help`) of the help message. Also in case
-//! of subcommands the short help message will automatically be used for the
-//! command description inside the parents help message and the long version
-//! as command description if help is requested on the actual subcommand.
-//!
-//! This feature can also be used with doc comments instead of arguments through
-//! proper comment formatting. To be activated it requires, that the first line
-//! of the comment is separated from the rest of the comment through an empty line.
-//! In this case the first line is used as summary and the whole comment represents
-//! the long descriptive message.
+//! For convenience, doc comments can be used instead of raw methods
+//! (this example works exactly as one above):
 //!
 //! ```
 //! # use structopt::StructOpt;
 //!
 //! #[derive(StructOpt)]
-//! /// The help message that will be displayed when passing `--help`.
+//! /// I am a program and I work, just pass `-h`
 //! struct Foo {
-//!   #[structopt(short)]
-//!   /// Only this summary is visible when passing `-h`.
-//!   ///
-//!   /// But the whole comment will be displayed when passing `--help`.
-//!   /// This could be quite useful to provide further hints are usage
-//!   /// examples.
+//!   /// Pass `-h` and you'll see me!
 //!   bar: String
 //! }
-//! # fn main() {}
 //! ```
 //!
+//! Doc comments on [top-level](#magical-methods) will be turned into
+//! `App::about/long_about` call (see below), doc comments on field-level are
+//! `Arg::help/long_help` calls.
+//!
+//! **Note:**
+//! _________________
+//!
+//! Raw methods have priority over doc comments!
+//!
+//! **Top level doc comments always generate `App::about/long_about` calls!**
+//! If you really want to use `App::help/long_help` method (you likely don't) use
```

Comment:
> If you really want to use **the** `App::help/long_help` method**s** (you likely don't)**,** use

I thought articles should be omitted before proper nouns (with some exceptions but I believe a name of a method isn't one of them).

Reply:
> Also, I just noticed the "with no doc comment" part. I'm averse to anything which requires constraining what can be fed to rustdoc.. Is there a way to force it to ignore the doc comment in this use case?

There's no such way, `App::about/long_about` call will be generated no matter what. But this is not a problem since [`App::help/long_help`](https://docs.rs/clap/2.33.0/clap/struct.App.html#method.help) will override it anyway. More specifically, `help` will override *the whole message*, including messages for arguments (but excluding messages for subcommands), see the `clap`'s doc.

Reply:
> I thought articles should be omitted before proper nouns (with some exceptions but I believe a name of a method isn't one of them).

As phrased, `App::help/long_help` is functioning as an adjective which modifies the plural noun "methods". "Methods" isn't a proper noun.

> There's no such way, App::about/long_about call will be generated no matter what. But this is not a problem since App::help/long_help will override it anyway. More specifically, help will override the whole message, including messages for arguments (but excluding messages for subcommands), see the clap's doc.

My point is that the documentation, as phrased, implies that you *must* leave the information rustdoc sees as blank to achieve the desired behaviour. I'm asking if that's correct or whether it needs to be revised to clearly explain how to override the doc comment so that it can still be used to feed rustdoc in that situation.

Reply:
> I'm asking if that's correct or whether it needs to be revised to clearly explain how to override the doc comment so that it can still be used to feed rustdoc in that situation.

It needs to be rephrased, definitely. I suggest

> use a raw method which will override the doc comment.

Reply:
I'd go with...

> If you really want to use the `App::help/long_help` methods (you likely don't)**, you can** use a raw method that will override the `App::about` call generated.

...but, otherwise, that sounds good.